docs/dev/development.md

```diff
@@ -33,26 +33,17 @@
 
 ## Developer quick start
 
-- Setup `GIT_HOST` to override the setting for your custom path.
+- Run the `code-dev` to check your code and run the unit test.
 
 ```bash
-export GIT_HOST=github.com/<YOUR_GITHUB_ID>
-```
-
-- Run the `linter` and `test` before building the binary.
-
-```bash
-make check
-make test
-make build
+make code-dev
 ```
 
 - Build and push the docker image for local development.
 
 ```bash
-export IMG=<YOUR_CUSTOMIZED_IMAGE_NAME>
-export REGISTRY=<YOUR_CUSTOMIZED_IMAGE_REGISTRY>
-make build-push-images
+export DEV_REGISTRY=<YOUR_CUSTOMIZED_IMAGE_REGISTRY>
```

Comment:
Do we need another registry variable? I think Makefile should only use one registry variable if possible.

Reply:
Yes, I think we need another one. Because the other two registries are used for quay.io and artifactory specifically.

Reply:
We probably should remove the quay one anyways since there is no reason for us to be manually pushing to quay.io/opencloudio. And PROW should always be using artifactory. Developers can just overwrite the artifactory registry variable when testing.

Reply:
Why don't we create a specific registry parameter for dev? I believe we need to prevent users from overwriting the artifactory registry which is used for CICD. 

We are using the quay.io for the e2e testing.

Reply:
The e2e testing should be changed then because it should not be using quay.io/opencloudio. That registry should be reserved for CICD to promote images to since accidental pushes to a release tag would cause problems.

Developers typically would not accidentally push to the integration Artifactory repo because of this check https://github.com/IBM/operand-deployment-lifecycle-manager/blob/master/Makefile#L67-L71. And even if they do, only users with permissions would be able to push there. Pushing to scratch repo is fine since that repo is meant for testing.

Reply:
The e2e testing is planned to be added to the CICD pipeline. Users won't have permission to push any images to the quay.io. 

I agree we should not push e2e test images, build a local image should be fine

Reply:
If you still have concern about the quay.io/opencloudio, we can open another quay.io registry for ODLM and installer components

Reply:
I agree we shouldn't push e2e images. I will update it in the following commit.

Reply:
Is there any reason the developer cannot just run
```
export ARTIFACTORYA_REGISTRY=<YOUR_CUSTOMIZED_IMAGE_REGISTRY>
```

Reply:
Yes, we can, but I believe `ARTIFACTORYA_REGISTRY` will confuse the developer. I believe `DEV_REGISTRY` is a parameter name that can explain itself.

Reply:
Yes, I agree ARTIFACTORYA_REGISTRY would be confusing, which is why I was wondering whether we can just change the name to `REGISTRY`

Reply:
@horis233 ^

Reply:
I think we should separate `DEV_REGISTRY` and `ARTIFACTORYA_REGISTRY`,  when developers export a common variable `registry` it will impact the other make commands. I don't see inconvenience caused by using `DEV_REGISTRY`. Did I miss something?

Reply:
@bitscuit If you have an idea about refactoring the Makefile, I am fine with it. 
But this PR focuses on making other opensource contributors try ODLM easily and update our documents.

Reply:
Yea, that makes sense. This PR should focus on one thing. I think we should try to keep variables to a minimum if possible, otherwise the Makefile gets more complex